src/main/java/me/drendov/HouseKeeping/Listeners.java

```diff
@@ -75,22 +79,51 @@ public void onPlayerToggleFlight(PlayerToggleFlightEvent event) {
         Player player = event.getPlayer();
         if (!player.isFlying()) return;
         if (!player.getWorld().getName().equals(HouseKeeping.getInstance().config.getSafezoneWorld())) return;
-        if (player.isOp() || player.hasPermission("housekeeping.bypass") || player.hasPermission("housekeeping.bypass.fly")) return;
+        if (player.isOp() || player.hasPermission("housekeeping.bypass") || player.hasPermission("housekeeping.bypass.fly"))
+            return;
 
         GameMode mode = player.getGameMode();
         if (mode != GameMode.CREATIVE && mode != GameMode.SPECTATOR) {
             HashMap getSafeZoneArea = HouseKeeping.getInstance().config.getSafeZoneArea(player);
             Location loc1 = (Location) getSafeZoneArea.get("loc1");
             Location loc2 = (Location) getSafeZoneArea.get("loc2");
-            if (! this.isInRect(player, loc1, loc2)) {
+            if (!this.isInRect(player, loc1, loc2)) {
                 HouseKeeping.sendMessage(player, TextMode.Err, Messages.CantFlyHere);
                 event.setCancelled(true);
                 player.setAllowFlight(false);
             }
         }
     }
 
-    public boolean isInRect(Player player, Location loc1, Location loc2) {
+    @EventHandler(priority = EventPriority.HIGHEST, ignoreCancelled = true)
+    public void onPortalLight(final PortalCreateEvent event) {
+
+        Player player = null;
+        if (event.getEntity() instanceof Player) {
+            player = (Player) event.getEntity();
+        }
+        if (player == null) {
+            event.setCancelled(true);
+            return;
+        }
+        if (player.isOp() || player.hasPermission("housekeeping.bypass") || player.hasPermission("housekeeping.bypass.portal"))
+            return;
+
+        HashMap getSafeZoneArea = HouseKeeping.getInstance().config.getSafeZoneArea(player);
+        Location loc1 = (Location) getSafeZoneArea.get("loc1");
+        Location loc2 = (Location) getSafeZoneArea.get("loc2");
+        if (this.isInRect(player, loc1, loc2)) {
+            if (event.getReason() == PortalCreateEvent.CreateReason.FIRE) {
+                if (HouseKeeping.getInstance().config.getRulePreventPortalCreation()) {
+                    HouseKeeping.sendMessage(player, TextMode.Err, Messages.DenyNetherPortalMsg);
```

Comment:
Should check not if player is in safe zone, but if portal (or block player tries to put on fire) is in safe zone. With current player-based check you can light a portal inside the safe zone by placing yourself outside.

Reply:
I'm aware of it. It's hard to do as for me)
having big safe zone (thousands of blocks), 2-3 non-safe blocks are OK unless this plugin will be published and we have to fix that.
raised #15 to address this issue later.